app/connectors/getPenaltyDetails/PenaltyDetailsConnector.scala

```diff
@@ -32,57 +35,148 @@ import java.time.LocalDateTime
 import java.util.UUID.randomUUID
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
+import connectors.parsers.getPenaltyDetails.PenaltyDetailsParser.PenaltyDetailsResponse
+import connectors.parsers.getPenaltyDetails.PenaltyDetailsParser.PenaltyDetailsFailureResponse
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import java.time.Instant
+import uk.gov.hmrc.http.HttpReads
```

Comment:
Minor things but lots of file warnings, can be cleaned up with things like Code > Optimise Imports, and adding a .scala.fmt.
'headerCarrier' vals are created but not used.

Reply:
![image](https://github.com/user-attachments/assets/8dd90686-ec38-44de-bf0d-6b94e27f9cff)